Tradução: program.qmd (#55)
cienciadedatos/pt-r4ds: program.qmd

# Programar {#sec-program-intro .unnumbered}

```{r}
#| results: "asis"
#| echo: false
source("_common.R")
```

Nessa parte do livro, você irá aprimorar suas habilidades em programação.

> Programação é uma competência transversal para todo trabalho de ciência de dados: você deve usar o computador para fazer ciência de dados; você não pode fazê-la em sua cabeça, com lápis e papel.

Reviewer: Apesar de a tradução literal não ser a mesma, achei que "essencial" faria mais sentido para o leitor em portugues. O que acham?

> Programação é uma competência transversal para todo trabalho de ciência de dados: você deve usar o computador para fazer ciência de dados; você não pode fazê-la em sua cabeça, com lápis e papel.

Author: Inicialmente tinha pensado por aí tb. Mas refletindo um pouco, eu acho que o que ele quer dizer é que a programação é uma competência que permeia todo o trabalho, no sentido de "percorrer" todo o processo / estar sempre presente. Acho que é um pouco mais além que "essencial", por isso mantive a tradução como "transversal". Faz sentido?

> Programação é uma competência transversal para todo trabalho de ciência de dados: você deve usar o computador para fazer ciência de dados; você não pode fazê-la em sua cabeça, com lápis e papel.

Reviewer: Entendi, mas acho que transversal não é um termo comum aplicado nestes casos pelo leitor em PT. Deixamos assim e quem sabe alguém aparece com alguma outra sugestão. 😉

> Programação é uma competência transversal para todo trabalho de ciência de dados: você deve usar o computador para fazer ciência de dados; você não pode fazê-la em sua cabeça, com lápis e papel.

Reviewer 2: Eu também entendi como a Lígia! É uma competência presente em todas as etapas

```{r}
#| label: fig-ds-program
#| echo: false
#| out.width: ~
#| fig-cap: |
#|   Programação é a água na qual todos os outros componentes nadam.
#| fig-alt: |
#|   Nosso modelo do processo de ciência de dados com a programação (importar,

> #|   arrumar, transformar, visualizar, modelar e comunicar, ou seja, tudo)

Reviewer: ```suggestion
#|   organizar, transformar, visualizar, modelar e comunicar, ou seja, tudo)
```

#|   destacada em azul.

knitr::include_graphics("diagrams/data-science/program.png", dpi = 270)
```

Programação produz código, e código é uma ferramenta de comunicação.
Obviamente o código diz ao computador o que você quer que ele faça.
Mas ele também transmite um significado a outros humanos.
Pensar em código como um veículo de comunicação é importante porque todo projeto que você faz é fundamentalmente colaborativo.
Mesmo que você não esteja trabalhando com outras pessoas, você definitivamente trabalhará com o você do futuro.

> Escrever códigos claros é importante para que outros (como você do futuro) possam entender o porquê de você ter abordado uma análise da forma como fez.

Reviewer 2: ```suggestion
Escrever códigos claros é importante para que outras pessoas (como você do futuro) possam entender o porquê de você ter abordado uma análise da forma como fez.
```

> Escrever códigos claros é importante para que outras pessoas (como você do futuro) possam entender o porquê de você ter abordado uma análise da forma como fez.

Reviewer 2: ```suggestion
Escrever códigos claros é importante para que outras pessoas (como você no futuro) possam entender o porquê de você ter abordado uma análise da forma como fez.
```

@scopinho você adicionou o "como **o** você", esse **o** é necessário?

Reviewer: Não sei dizer se é necessário, mas pelas aparições no google (procurar por "o você do futuro" e "você do futuro", esse **o** aparece mais nos textos, apesar se não ver problema sem deixarmos sem também.

Isso significa que melhorar na programação também envolve melhorar na comunicação.
Com o tempo, você quer que seu código seja não apenas mais fácil de escrever, mas fácil também para outras pessoas lerem.

Nos próximos três capítulos, você irá desenvolver competências que irão aprimorar suas habilidades em programação.

1.  Copiar e colar é uma ferramenta poderosa, mas você deve evitar usá-la mais que duas vezes.
    Código repetitivo é perigoso, pois ele pode facilmente te levar a erros e inconsistências.
    Em vez disso, no @sec-functions, você irá aprender como escrever **funções** que te permitem extrair códigos tidyverse repetidos para que possam ser facilmente reutilizados.

2.  Funções extraem os códigos repetidos, mas você eventualmente precisa repetir as mesmas ações em diferentes *inputs*.
    Você precisa de ferramentas de **iteração** que te permitem executar as mesmas ações de novo e de novo.
    Essas ferramentas incluem *for loops* e programação funcional, que você irá aprender no @sec-iteration.

3.  Quanto mais você ler códigos escritos por outras pessoas, mais você verá códigos que não utilizam tidyverse.

>     No @sec-base-r, você irá aprender algumas das funções mais importantes do R base que você vê por aí.

Reviewer 2: ```suggestion
    No @sec-base-r, você irá aprender algumas das funções mais importantes do R base que você verá por aí.
```

O objetivo desses capítulos é te ensinar o mínimo sobre programação que você precisa para ciência de dados.

> Quando você dominar esse material, recomendamos fortemente que você continue a investir em suas habilidades de programação.

Reviewer 2: ```suggestion
Quando você dominar esse material, recomendamos fortemente que você continue a investir nas suas habilidades de programação.
```

Nós escrevemos dois livros que você pode achar útil.

> [*Hands on Programming with R*](https://rstudio-education.github.io/hopr/), de Garrett Grolemund, é uma introdução a R como linguagem de programação e um ótimo ponto de partida se R é sua primeira linguagem de programação.

Reviewer: ```suggestion
[*Hands on Programming with R*](https://rstudio-education.github.io/hopr/), de Garrett Grolemund, é uma introdução ao R como linguagem de programação e um ótimo ponto de partida se o R for sua primeira linguagem de programação.
```

> [*Advanced R*](https://adv-r.hadley.nz/) de Hadley Wickham entra em detalhes na linguagem de programação R; é um ótimo ponto de partida se você já tem experiência em programação e ótimo próximo passo se você já tiver internalizado as ideias desses capítulos.

Reviewer: ```suggestion
[*Advanced R*](https://adv-r.hadley.nz/) de Hadley Wickham entra em detalhes na linguagem de programação R; é um ótimo ponto de partida se você já tem experiência em programação e ótimo próximo passo se você já estiver familiarizado com as ideias desses capítulos.
```

> [*Advanced R*](https://adv-r.hadley.nz/) de Hadley Wickham entra em detalhes na linguagem de programação R; é um ótimo ponto de partida se você já tem experiência em programação e ótimo próximo passo se você já tiver internalizado as ideias desses capítulos.

Reviewer 2: ```suggestion
[*Advanced R*](https://adv-r.hadley.nz/) de Hadley Wickham entra em detalhes na linguagem de programação R; é um ótimo ponto de partida se você já tem experiência em programação e ótimo próximo passo se as ideias desses capítulos forem familiares para você.
```

Tentando reescrever sem marcar o gênero na palavra "familiarizad**o**". acham que a leitura está ok? @ligialouzada @scopinho

Reviewer: Faz sentido sim @beatrizmilz
